modules/twinkleblock.js

```diff
@@ -477,6 +477,15 @@ Twinkle.block.callback.change_action = function twinkleblockCallbackChangeAction
 
 		$form.find('[name=namespacerestrictions]').select2({
 			width: '100%',
+			matcher: function(params, data) {
```

Comment:
Very nice!  Much better, this has been bugging me.

Reply:
Should we put this in morebits? This is very much reusable.

Reply:
If we're not going to reuse it not a huge need, but couldn't hurt; should be broadly applicable enough.

Reply:
Done.